src/compiler/crystal/syntax/parser.cr

```diff
@@ -5662,18 +5662,24 @@ module Crystal
       check :","
 
       next_token_skip_space_or_newline
-      check :INSTANCE_VAR
-
-      ivar_location = @token.location
-      instance_var = InstanceVar.new(@token.value.to_s).at(ivar_location)
+      offset = case @token.type
+               when :INSTANCE_VAR
+                 InstanceVar.new(@token.value.to_s)
+               when :NUMBER
+                 raise "expecting an integer offset, not '#{@token}'", @token if @token.number_kind != :i32
```

Comment:
I'd use a class name instead.

Reply:
I didn't get it., Do you mean write this as `raise Whatever.new("...")`? I wrote `raise "..."` just to follow the way other places in the parser do when an error happen.

Reply:
No, I mean using a class/type within the error message and not the value, ie.

```suggestion
                 raise "expecting an integer offset, not '#{@token.class}'", @token if @token.number_kind != :i32
```

Reply:
But that's wrong...

Reply:
@token is always a Token

Reply:
But it has a `#kind` property or sth like that, no?

Reply:
`Token#to_s` handles presentation

Reply:
Fair enough, thanks for bearing with me 🙏